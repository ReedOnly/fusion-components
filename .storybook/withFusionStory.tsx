import React from 'react';
import Header from '../src/components/core/Header';
import Content from '../src/components/core/Content';
import Overlay from '../src/components/general/ApplicationGuidance/components/Overlay';

export default (title: string, padding: number = 16) => (stories: () => React.ReactNode) => {
    return (
        <>
            <Overlay fixed>
                <Header aside={null} start={null} content={null} />
            </Overlay>
            <Content>
                <Overlay>
                    <div
                        style={{
                            padding,
                            minHeight: '100%',
                            width: '100%',
                            maxWidth: '100%',
                            boxSizing: 'border-box',
                        }}
                    >
                        {stories()}
                    </div>
                </Overlay>
            </Content>
        </>
    );
};
