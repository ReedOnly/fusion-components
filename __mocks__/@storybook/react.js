import renderer from 'react-test-renderer';
import addons, { mockChannel } from '@storybook/addons';
import { withFusionContext } from '../../.storybook/withFusionContext';

addons.setChannel(mockChannel());

export const action = (actionName) => jest.fn();
const fusionContext = withFusionContext();
export const storiesOf = (groupName) => {
    const api = {
        decorators: [],
        parameters: {},

        add(storyName, story) {
            if (api.parameters.skipTest) {
                return;
            }

            describe(groupName, () => {
                it(storyName, () => {
                    // const decoratedStory = api.decorators.reduce((s, decorator) => {
                    //     try {
                    //         return decorator(s, { parameters: api.parameters });
                    //     } catch (e) {
                    //         return s;
                    //     }
                    // }, story);

                    const component = renderer.create(fusionContext(story));

                    // @TODO Enable custom names once released:
                    // > https://github.com/facebook/jest/pull/2094
                    expect(component.toJSON())
                        .toMatchSnapshot
                        // `${groupName}.${storyName}`
                        ();
                });
            });

            return api;
        },

        addDecorator() {
            return api;
        },

        addParameters(parameters) {
            api.parameters = { ...api.parameters, ...parameters };
            return api;
        },
    };

    return api;
};

export const addDecorator = () => {};
