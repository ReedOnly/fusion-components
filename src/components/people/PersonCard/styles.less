.container {
    padding: calc(var(--grid-unit) * 1px);
    display: flex;
    align-items: center;

    .details {
        display: flex;
        flex-direction: column;
        padding-left: calc(var(--grid-unit) * 2px); 

        .name {
            padding-bottom: calc(var(--grid-unit) * .5px);
        }

        .email {
            padding-top: calc(var(--grid-unit) * .5px);
        }
    }
}