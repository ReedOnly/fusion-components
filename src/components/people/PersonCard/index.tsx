import React, { useState, useEffect } from 'react';
import { PersonPhoto, PhotoSize } from '@equinor/fusion-components';
import classNames from 'classnames';
import styles from './styles.less';
import { PersonDetails, usePersonDetails, useComponentDisplayClassNames } from '@equinor/fusion';
import { getDefaultPerson } from '../utils';

export { PhotoSize };

export type PersonCardProps = {
    personId?: string;
    person?: PersonDetails;
    photoSize?: PhotoSize;
};

export default ({ personId, person, photoSize = 'xlarge' }: PersonCardProps) => {
    const [currentPerson, setCurrentPerson] = useState<PersonDetails>(getDefaultPerson());
    const [error, _, personDetails] = personId ? usePersonDetails(personId) : [null, false, person];

    useEffect(() => {
        if (!error && personDetails) {
            setCurrentPerson(personDetails);
        } else if (error) {
            setCurrentPerson(getDefaultPerson());
        }
    }, [error, personDetails]);

    const containerClassNames = classNames(styles.container, useComponentDisplayClassNames(styles));
    const detailsClassNames = classNames(styles.details, useComponentDisplayClassNames(styles));
    const nameClassNames = classNames(styles.name, useComponentDisplayClassNames(styles));
    const emailClassNames = classNames(styles.email, useComponentDisplayClassNames(styles));

    return (
        <div className={containerClassNames}>
            <PersonPhoto personId={currentPerson.azureUniqueId} size={photoSize} />
            <div className={detailsClassNames}>
                <div className={nameClassNames}>{currentPerson.name}</div>
                <div className={emailClassNames}>
                    <a href={`mailto:${currentPerson.mail}`}>{currentPerson.mail}</a>
                </div>
            </div>
        </div>
    );
};
