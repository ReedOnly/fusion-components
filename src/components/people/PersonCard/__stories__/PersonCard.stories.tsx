import React from 'react';
import { storiesOf } from '@storybook/react';
import withFusionStory from '../../../../../.storybook/withFusionStory';
import PersonCard from '../index';

storiesOf('People|PersonCard', module)
    .addParameters({ jest: ['PersonCard.stories.jsx'] })
    .addDecorator(withFusionStory('PersonCard'))
    .add('Default', () => (
        <PersonCard
            personId="e92c631b-94ae-4670-8f1e-60cdc2122edc"
            personName="Morten Salte"
            email="msal@equinor.com"
            photoSize="large"
        />
    ));
