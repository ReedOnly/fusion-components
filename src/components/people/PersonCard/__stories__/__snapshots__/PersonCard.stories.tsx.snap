// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`People|PersonCard Default 1`] = `
<div
  className="container comfortable"
>
  <div
    className="container comfortable"
  >
    <div
      style={
        Object {
          "paddingBottom": "16px",
        }
      }
    >
      <div />
    </div>
    <div
      style={
        Object {
          "paddingBottom": "16px",
        }
      }
    >
      <div
        className="container comfortable"
      >
        <div
          className="photoContainer comfortable xlarge"
          style={
            Object {
              "backgroundImage": "url()",
            }
          }
        >
          <div>
            <div
              className="popoverDetails"
            >
              <svg
                height={56}
                style={
                  Object {
                    "color": "",
                  }
                }
                viewBox="0 0 56 56 "
                width={56}
              >
                <circle
                  cx="28"
                  cy="28"
                  fill="#EAEAEA"
                  r="28"
                />
                <path
                  clipRule="evenodd"
                  d="M28.0001 14.6666C24.3167 14.6666 21.3334 17.65 21.3334 21.3333C21.3334 25.0166 24.3167 28 28.0001 28C31.6834 28 34.6667 25.0166 34.6667 21.3333C34.6667 17.65 31.6834 14.6666 28.0001 14.6666ZM31.3334 21.3333C31.3334 19.5 29.8334 18 28.0001 18C26.1667 18 24.6667 19.5 24.6667 21.3333C24.6667 23.1666 26.1667 24.6666 28.0001 24.6666C29.8334 24.6666 31.3334 23.1666 31.3334 21.3333ZM38.0001 38C37.6667 36.8166 32.5001 34.6666 28.0001 34.6666C23.5167 34.6666 18.3834 36.8 18.0001 38H38.0001ZM14.6667 38C14.6667 33.5666 23.5501 31.3333 28.0001 31.3333C32.4501 31.3333 41.3334 33.5666 41.3334 38V41.3333H14.6667V38Z"
                  fill="#666666"
                  fillRule="evenodd"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="details"
        >
          <div
            className="name"
          >
            <div
              className="bar comfortable skeleton"
              style={
                Object {
                  "height": "calc(var(--grid-unit) * 1.5px)",
                  "width": "50%",
                }
              }
            />
          </div>
          <div
            className="jobTitle"
          >
            <div
              className="bar comfortable skeleton"
              style={
                Object {
                  "height": "calc(var(--grid-unit) * 1.5px)",
                  "width": "50%",
                }
              }
            />
          </div>
          <div>
            <div
              className="bar comfortable skeleton"
              style={
                Object {
                  "height": "calc(var(--grid-unit) * 1.5px)",
                  "width": "50%",
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        className="container comfortable"
      >
        <div
          className="photoContainer comfortable xlarge"
          style={
            Object {
              "backgroundImage": "url()",
            }
          }
        >
          <div>
            <div
              className="popoverDetails"
            >
              <svg
                height={56}
                style={
                  Object {
                    "color": "",
                  }
                }
                viewBox="0 0 56 56 "
                width={56}
              >
                <circle
                  cx="28"
                  cy="28"
                  fill="#EAEAEA"
                  r="28"
                />
                <path
                  clipRule="evenodd"
                  d="M28.0001 14.6666C24.3167 14.6666 21.3334 17.65 21.3334 21.3333C21.3334 25.0166 24.3167 28 28.0001 28C31.6834 28 34.6667 25.0166 34.6667 21.3333C34.6667 17.65 31.6834 14.6666 28.0001 14.6666ZM31.3334 21.3333C31.3334 19.5 29.8334 18 28.0001 18C26.1667 18 24.6667 19.5 24.6667 21.3333C24.6667 23.1666 26.1667 24.6666 28.0001 24.6666C29.8334 24.6666 31.3334 23.1666 31.3334 21.3333ZM38.0001 38C37.6667 36.8166 32.5001 34.6666 28.0001 34.6666C23.5167 34.6666 18.3834 36.8 18.0001 38H38.0001ZM14.6667 38C14.6667 33.5666 23.5501 31.3333 28.0001 31.3333C32.4501 31.3333 41.3334 33.5666 41.3334 38V41.3333H14.6667V38Z"
                  fill="#666666"
                  fillRule="evenodd"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="details"
        >
          <div
            className="name"
          >
            <div
              className="bar comfortable skeleton"
              style={
                Object {
                  "height": "calc(var(--grid-unit) * 1.5px)",
                  "width": "50%",
                }
              }
            />
          </div>
          <div
            className="jobTitle"
          >
            <div
              className="bar comfortable skeleton"
              style={
                Object {
                  "height": "calc(var(--grid-unit) * 1.5px)",
                  "width": "50%",
                }
              }
            />
          </div>
          <div>
            <div
              className="bar comfortable skeleton"
              style={
                Object {
                  "height": "calc(var(--grid-unit) * 1.5px)",
                  "width": "50%",
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="overlay"
  />
  <div
    className="snacks"
  />
  <div
    className="dialog"
  />
</div>
`;
